File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.java
class Solution {
    public int[] leftRightDifference(int[] nums) {
        int n = nums.length;
        int[] leftSum = new int[n];
        leftSum[0] = 0;
        int[] rightSum = new int[n];
        rightSum[n-1] = 0;
        int sum = 0;
        for(int i=0;i<n-1;i++){
            sum+=nums[i];
            leftSum[i+1]=sum;
        }
        sum = 0;
        for(int i=n-1;i>0;i--){
            sum+=nums[i];
            rightSum[i-1] = sum;
        }
        int[] res = new int[n];
        for(int i=0;i<n;i++){
            res[i] = Math.abs(leftSum[i] - rightSum[i]);
        }
        return res;
    }
}

File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var leftRightDifference = function(nums) {
    let n = nums.length;
    if(n === 1){
        return [0];
    }
    let left = new Array(n);
    left[0] = 0;
    let right = new Array(n);
    right[n-1] = 0;
    let sum = 0;
    for(let i = 1;i<n;i++){
        sum+=nums[i-1];
        left[i] = sum;
    }
    sum = 0;
    for(let i = n-2;i>=0;i--){
        sum+=nums[i+1];
        right[i] = sum;
    }
    let ans = [];
    for(let i = 0;i<left.length;i++){
        ans.push(Math.abs(left[i]-right[i]));
    }
    return ans;
};
